tests/test_memory.py

```diff
@@ -279,3 +280,28 @@ def test_memory_buffer_bytearray():
     assert len(byte_array) == 1114112
     assert byte_array[0:3] == b'\x01\x02\x03'
     assert byte_array[3:9].decode() == 'Wasmer'
+
+def test_memory_buffer_supports_ctypes():
+    c_uint8_4 = ctypes.c_uint8 * 4
+
+    memory = instance().exports.memory
+
+    arr = c_uint8_4.from_buffer(memory.buffer)
+    arr[0] = 0b00000001
+    arr[1] = 0b00000100
+    arr[2] = 0b00010000
+    arr[3] = 0b01000000
+
+    byte_array = bytearray(memory.buffer)
+    assert byte_array[0] == 0b00000001
+    assert byte_array[1] == 0b00000100
+    assert byte_array[2] == 0b00010000
+    assert byte_array[3] == 0b01000000
+
+def test_memory_buffer_supports_keeps_object_alive():
+    # IMPL NOTE: make sure the memory instance is immediately del'd
+    view = memoryview(instance().exports.memory.buffer)
+
+    val = bytes([42] * 1024)
+    for i in range(len(view) // 1024):
+        view[i * 1024:(i + 1) * 1024] = val
```

Comment:
What does it check?

Reply:
The idea was to write to memory, that is only valid to write to, if the ownership semantics are correctly expressed. Probably, not a very good test, as it is intederministic in nature outside of a memory debugger such as valgrind. I.e., depending on your system, it could pass even, though you completely destroy non-owned memory

At least on my system, it immediately segfeaulted with the previous ownership setup (`(*view).obj = ptr::null_mut();`). 

TBH. I'm not sure it makes sense to keep it. I'll add a docstring explaining the motivation behind the test, but I'm happy to remove it. In any case, it was helpful for me to verify that there was indeed an issue.

Reply:
I'm happy to keep the tests to avoid regression. Please add a documentation explaining what it showcases even if it doesn't test exactly what we want. That's a valid regression test after all :-)!